test/Shared/UserManagerTestBase.cs

```diff
@@ -139,18 +134,17 @@ protected TRole CreateRole(string namePrefix = "")
         public async Task CanUpdatePasswordUsingHasher()
         {
             var manager = CreateManager();
-            var user = new TUser() { UserName = "UpdatePassword" };
+            var user = CreateTestUser("UpdatePassword");
             IdentityResultAssert.IsSuccess(await manager.CreateAsync(user, "password"));
             Assert.True(await manager.CheckPasswordAsync(user, "password"));
-            string expectedLog = string.Format("{0} for user: {1} : {2}", "CheckPasswordAsync", user.Id.ToString(), true.ToString());
+            string expectedLog = string.Format("{0} for user: {1} : {2}", "CheckPasswordAsync", await manager.GetUserIdAsync(user), true.ToString());
             IdentityResultAssert.VerifyLogMessage(manager.Logger, expectedLog);
 
-            user.PasswordHash = manager.PasswordHasher.HashPassword(user, "New");
-            IdentityResultAssert.IsSuccess(await manager.UpdateAsync(user));
+            IdentityResultAssert.IsSuccess(await manager.ChangePasswordAsync(user, "password", "Password1!"));
             Assert.False(await manager.CheckPasswordAsync(user, "password"));
-            expectedLog = string.Format("{0} for user: {1} : {2}", "CheckPasswordAsync", user.Id.ToString(), false.ToString());
+            expectedLog = string.Format("{0} for user: {1} : {2}", "CheckPasswordAsync", await manager.GetUserIdAsync(user), false.ToString());
             IdentityResultAssert.VerifyLogMessage(manager.Logger, expectedLog);
-            Assert.True(await manager.CheckPasswordAsync(user, "New"));
+            Assert.True(await manager.CheckPasswordAsync(user, "Password1!"));
```

Comment:
If I am reading this correctly, it sounds very complicated, and it won't work :)

http://blogs.msdn.com/b/ericlippert/archive/2012/11/13/why-is-deriving-a-public-class-from-an-internal-class-illegal.aspx

I think it is actually fine to use the API on user manager as an abstraction layer and if we just need to do some trick to set the password hash alone, we can look into that more.


-------- Original message --------
From: Suhas Joshi <notifications@github.com>
Date: 03/03/2015 19:31 (GMT-08:00)
To: aspnet/Identity <Identity@noreply.github.com>
Cc: Diego Vega <Diego.Vega@microsoft.com>
Subject: Re: [Identity] [Feedback PR] Moved POCOs to EF and fixed tests (#378)


In test/Shared/UserManagerTestBase.cs<https://github.com/aspnet/Identity/pull/378#discussion_r25749952>:

>              IdentityResultAssert.VerifyLogMessage(manager.Logger, expectedLog);
> -            Assert.True(await manager.CheckPasswordAsync(user, "New"));
> +            Assert.True(await manager.CheckPasswordAsync(user, "Password1!"));


I like this solution. I'll have IdentityUserBase internal in Identity project all except the navigation properties. EF.IdentityUser extends this to add navigation properties and TestUser extends this to add properties as needed.

There is also an ApplicationUser in the test shared folder. Can I replace that too ?

-
Reply to this email directly or view it on GitHub<https://github.com/aspnet/Identity/pull/378/files#r25749952>.